tests/integration/e2e/test_basic_flow.py

```diff
@@ -47,19 +50,86 @@ async def test_cluster_is_deployed_successfully(
         ),
     )
     await ops_test.model.wait_for_idle(
-        apps=[KAFKA_CHARM_NAME, ZOOKEEPER_CHARM_NAME, DATABASE_CHARM_NAME], status="active"
+        apps=[KAFKA_CHARM_NAME, ZOOKEEPER_CHARM_NAME, DATABASE_CHARM_NAME],
+        status="active",
+        timeout=1200,
+        idle_period=30,
     )
 
 
 @pytest.mark.abort_on_fail
-async def test_test_app_actually_set_up(ops_test: OpsTest, deploy_test_app):
-    # deploy producer and consumer
+async def test_test_app_actually_set_up(
+    ops_test: OpsTest, deploy_test_app, kafka, deploy_data_integrator, integrator
+):
+    # consumer and producer params when deploying with data-integrator
+    producer_parameters = None
+    consumer_parameters = None
+    if integrator:
+        # deploy integrators and get credentials
+        data_integrator_producer = await deploy_data_integrator(
+            {"topic-name": TOPIC, "extra-user-roles": "producer"}
+        )
+
+        await ops_test.model.add_relation(data_integrator_producer, kafka)
+        await ops_test.model.wait_for_idle(
+            apps=[data_integrator_producer, kafka], idle_period=30, status="active", timeout=1800
+        )
+
+        producer_credentials = await fetch_action_get_credentials(
+            ops_test.model.applications[data_integrator_producer].units[0]
+        )
+        producer_parameters = get_action_parameters(producer_credentials, TOPIC)
+
+        data_integrator_consumer = await deploy_data_integrator(
+            {"topic-name": TOPIC, "extra-user-roles": "consumer"}
+        )
+        await ops_test.model.add_relation(data_integrator_consumer, kafka)
+        await ops_test.model.wait_for_idle(
+            apps=[data_integrator_consumer, kafka], idle_period=30, status="active", timeout=1800
+        )
+        consumer_credentials = await fetch_action_get_credentials(
+            ops_test.model.applications[data_integrator_consumer].units[0]
+        )
+        consumer_parameters = get_action_parameters(consumer_credentials, TOPIC)
+        assert producer_parameters != consumer_parameters
 
+    # deploy producer and consumer
     producer = await deploy_test_app(role="producer", topic_name=TOPIC)
     assert ops_test.model.applications[producer].status == "active"
+
+    if integrator:
+        # start producer with action
+        assert producer_parameters
+        pid = await fetch_action_start_process(
+            ops_test.model.applications[producer].units[0], producer_parameters
+        )
```

Comment:
**suggestion:** Here I think we should loop over `units` rather than specifying each indexed unit.